packages/bot-web-ui/src/utils/journal-notifications.js

```diff
@@ -0,0 +1,62 @@
+import { localize } from '@deriv/translations';
+import { messageWithButton, arrayAsMessage } from '../components/notify-item.jsx';
+
+const showErrorMessageWithButton = (message, block_id, showErrorMessage, centerAndHighlightBlock) => {
+    showErrorMessage(
+        messageWithButton({
+            unique_id: block_id,
+            type: 'error',
+            message,
+            btn_text: localize('Go to block'),
+            onClick: () => {
+                centerAndHighlightBlock();
+            },
+        })
+    );
+};
+
+export const validateJournalMessage = (
+    { message, block_id, variable_name },
+    showErrorMessage,
+    centerAndHighlightBlock,
+    pushMessage
+) => {
+    // notify undefined variable block
+    if (message === undefined && variable_name != null) {
```

Comment:
Users can also have `undefined` values as part of a list, this should be checked too. Currently the page will crash if a user has an `undefined` element in their array.